x-pack/plugins/canvas/public/components/workpad_header/share_menu/flyout/flyout.tsx

```diff
@@ -0,0 +1,107 @@
+
+import {
+  getWorkpad,
+  getRenderedWorkpad,
+  getRenderedWorkpadExpressions,
+} from '../../../../state/selectors/workpad';
+
+import { ShareWebsiteFlyout as FlyoutComponent } from './flyout.component';
+import { State, CanvasWorkpad } from '../../../../../types';
+import { CanvasRenderedWorkpad } from '../../../../../shareable_runtime/types';
+import { renderFunctionNames } from '../../../../../shareable_runtime/supported_renderers';
+
+import { OnCloseFn } from '../share_menu.component';
+import { useNotifyService } from '../../../../services';
+
+import { useDownloadRenderedWorkpad } from '../../../hooks';
+import { useDownloadRuntime, useDownloadZippedRuntime } from './hooks';
+
+export { OnDownloadFn, OnCopyFn } from './flyout.component';
+
+const strings = {
+  getCopyShareConfigMessage: () =>
+    i18n.translate('xpack.canvas.workpadHeaderShareMenu.copyShareConfigMessage', {
+      defaultMessage: 'Copied share markup to clipboard',
+    }),
+  getUnknownExportErrorMessage: (type: string) =>
+    i18n.translate('xpack.canvas.workpadHeaderShareMenu.unknownExportErrorMessage', {
+      defaultMessage: 'Unknown export type: {type}',
+      values: {
+        type,
+      },
+    }),
+};
+
+const getUnsupportedRenderers = (state: State) => {
+  const renderers: string[] = [];
+  const expressions = getRenderedWorkpadExpressions(state);
+  expressions.forEach((expression) => {
+    if (!renderFunctionNames.includes(expression)) {
+      renderers.push(expression);
+    }
+  });
+
+  return renderers;
+};
+
+interface Props {
+  onClose: OnCloseFn;
+  renderedWorkpad: CanvasRenderedWorkpad;
+  unsupportedRenderers: string[];
+  workpad: CanvasWorkpad;
+}
+
+export const ShareWebsiteFlyout: FC<Pick<Props, 'onClose'>> = ({ onClose }) => {
+  const notifyService = useNotifyService();
+  const downloadRenderedWorkpad = useDownloadRenderedWorkpad();
+  const downloadRuntime = useDownloadRuntime();
+  const downloadZippedRuntime = useDownloadZippedRuntime();
+
+  const { renderedWorkpad, unsupportedRenderers } = useSelector((state: State) => ({
+    renderedWorkpad: getRenderedWorkpad(state),
+    unsupportedRenderers: getUnsupportedRenderers(state),
+    workpad: getWorkpad(state),
+  }));
+
+  const onCopy = useCallback(() => notifyService.info(strings.getCopyShareConfigMessage()), [
+    notifyService,
+  ]);
+
+  const onDownload = useCallback(
+    (type: string) => {
+      switch (type) {
+        case 'share':
+          downloadRenderedWorkpad(renderedWorkpad);
+          return;
+        case 'shareRuntime':
+          downloadRuntime();
+          return;
+        case 'shareZip':
+          downloadZippedRuntime(renderedWorkpad);
+          return;
+        default:
+          throw new Error(strings.getUnknownExportErrorMessage(type));
+      }
+    },
+    [downloadRenderedWorkpad, downloadRuntime, downloadZippedRuntime, renderedWorkpad]
+  );
+
+  return (
+    <FlyoutComponent
+      onCopy={onCopy}
```

Comment:
I'll need to take this into account in https://github.com/elastic/kibana/pull/104113